### PR TITLE
Add effort section in Developer Guide

### LINKED ISSUE #299: DG Effort page

### PR DESCRIPTION
**Pull request by Md-Fazil**
1. Add a section in the Appendix to detail the contributions and challenges faced by the team.

Closes #299.

### CODE REVIEW

**Review by github-amanda — APPROVED** [review]
LGTM!

**Review by chewwh09 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1110,3 +1110,47 @@ testers are expected to do more *exploratory* testing.
...
+
+4. Implementing the reminder and sort features required a deep understanding of the design of AB3. Merely manipulating 
+   variables in existing code was not sufficient, and we had to create many new classes such as `ReminderCommand`, 
+   `ReminderCommandParser`, `SortAscendingQuantityCommand` and etc. We also had to come up with ways to keep track of 
+   the state of the currently displayed list, to allow users to work with a sublist of items instead of the entire 
+   inventory of items. There was not a need for this in AB3.
+
+### Working Process
+
+Our team placed heavy emphasis on communication, and efficient division of workload. We made use of github's issue trackers, which allowed efficient division and tracking of work, and also helped each
```
Comment (kumsssss) [inline docs/DeveloperGuide.md]: Idt need the comma after the communication.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1110,3 +1110,47 @@ testers are expected to do more *exploratory* testing.
...
+meetings. Despite the unprecedented circumstances, our team persevered and developed a product that we are all proud of.
+
+### Major Enhancements
+
+StoreMando has many significant enhancements from AB3. Here are some examples:
+
+1. There was a need to change a large portion of the AB3 code, as we had a completely different set of variables for 
+   StoreMando. We had to do a lot of refactoring in the form of renaming class and attributes. The restrictions on 
+   fields in AB3 were also different from those in StoreMando (eg. Phone in AB3 must have 8 digits whereas Quantity in 
+   StoreMando can only have values within the range 1 to 1000000), and we had to edit those restrictions respectively.
```
Comment (kumsssss) [inline docs/DeveloperGuide.md]: Maybe can talk about changing compulsory Email attribute to optional Expiry Date. Just to have more things to mention.

Comment (Md-Fazil) [reply docs/DeveloperGuide.md]: Noted and fixed!

**Pull request merged**